homeassistant/components/sensor/citybikes.py

```diff
@@ -0,0 +1,310 @@
+                hass, latitude, longitude)
+            if not network_id:
+                async_track_point_in_utc_time(hass, async_setup_network,
+                                              dt_util.utcnow() + SCAN_INTERVAL)
+                return
+
+        if network_id not in MONITORED_NETWORKS:
+            network = CityBikesNetwork(hass, network_id)
+            MONITORED_NETWORKS[network_id] = network
+            hass.async_add_job(network.async_refresh)
+            async_track_time_interval(hass, network.async_refresh,
+                                      SCAN_INTERVAL)
+        else:
+            network = MONITORED_NETWORKS[network_id]
+
+        while not network.ready:
+            yield from asyncio.sleep(0.5)
+
+        entities = []
+        for station in network.stations:
+            dist = location.distance(latitude, longitude,
+                                     station[ATTR_LATITUDE],
+                                     station[ATTR_LONGITUDE])
+            station_id = station[ATTR_ID]
+            station_uid = str(station.get(ATTR_EXTRA, {}).get(ATTR_UID, ''))
+
+            if radius > dist or stations_list.intersection((station_id,
+                                                            station_uid)):
+                entities.append(CityBikesStation(network, station_id, name))
+
+        async_add_entities(entities, True)
+
+    hass.async_add_job(async_setup_network)
+
+
+class CityBikesNetwork:
+    """Thin wrapper around a CityBikes network object."""
+
+    @classmethod
+    @asyncio.coroutine
+    def get_closest_network_id(cls, hass, latitude, longitude):
+        """Return the id of the network closest to provided location."""
+        try:
+            session = async_get_clientsession(hass)
+            with async_timeout.timeout(5, loop=hass.loop):
+                req = yield from session.get(DEFAULT_ENDPOINT.format(
+                    uri=NETWORKS_URI))
+            json_response = yield from req.json()
+            networks = NETWORKS_RESPONSE_SCHEMA(json_response)
+            network = networks[ATTR_NETWORKS_LIST][0]
+            result = network[ATTR_ID]
+            minimum_dist = location.distance(
+                latitude, longitude,
+                network[ATTR_LOCATION][ATTR_LATITUDE],
+                network[ATTR_LOCATION][ATTR_LONGITUDE])
+            for network in networks[ATTR_NETWORKS_LIST][1:]:
+                network_latitude = network[ATTR_LOCATION][ATTR_LATITUDE]
+                network_longitude = network[ATTR_LOCATION][ATTR_LONGITUDE]
+                dist = location.distance(latitude, longitude,
+                                         network_latitude, network_longitude)
+                if dist < minimum_dist:
+                    minimum_dist = dist
+                    result = network[ATTR_ID]
+
+            return result
+        except (asyncio.TimeoutError, aiohttp.ClientError):
```

Comment:
It seems that `aiohttp.ClientError` is an umbrella exception to all the possible errors that can happen when performing an HTTP request, including any server errors: http://aiohttp.readthedocs.io/en/stable/client_reference.html#client-exceptions